app/src/main/AndroidManifest.xml

```diff
@@ -61,7 +62,8 @@
         </activity>
         <activity
             android:name=".app.FilterEditorActivity"
-            android:label="@string/filter_editor">
+            android:label="@string/filter_editor"
+            android:exported="true">
```

Comment:
Same - this should be only visible internally

Reply:
Firstly I bump targetSdkVersion to 31, and then it can't be compiled since if your app targets Android 12 or higher and contains activities, services, or broadcast receivers that use intent filters, you must explicitly declare the android:exported attribute for these app components. For more information you can read https://developer.android.com/about/versions/12/behavior-changes-12#exported.
For the MainActivity and FilterEditorActivity, if I set android:exported to false the Android Studio will warn me that a launchable activity must be exported as of Android 12, which also makes it available to other apps. So I think it should be exported.
For BlockedSmsReceiver, I'm not sure if it should be exported. Since I see broadcastBlockedSms in SmsHandlerHook, I think it should be exported to receive that broadcast.

Reply:
Ah ok I didn't know about this change, makes sense for the main activity. Still not sure why the filter editor activity needs to be exported though - it's not supposed to be accessed through the launcher.

The `BlockedSmsReceiver` change looks correct to me (I'm surprised it worked before without it actually - maybe system processes have some special permissions)

Reply:
Oh I'm dumb, for some reason `FilterEditorActivity` has a `android.intent.action.MAIN` intent filter... I think that one was probably cargo-cult copied from somewhere - we should just delete that intent filter since it's only ever started internally via explicit intent.